library/modify_yaml.py

```diff
@@ -20,6 +20,23 @@
     yaml_value: 2
 '''
 
+
+def set_key(yaml_data, yaml_key, yaml_value):
+    changes = []
+    ptr = yaml_data
+    for key in yaml_key.split('.'):
+        if key not in ptr and key != yaml_key.split('.')[-1]:
+            ptr[key] = {}
+            ptr = ptr[key]
```

Comment:
This line above was the bugfix, needed to re-assign the ptr after creating a new section.